impress-org/givewp
# Milestone can now be associated with one or many forms

_**Note: This PR is depended upon work in #5224 - it cannot be merged until that PR is first approved and merged!**_

Resolves #5226

## Description
This PR introduces functionality to associate one or many different forms with a Milestone block. Previously, no attributes or fields existed to manage which form IDs are used to render a given Milestone. To address this, the `ids` attributes was introduced, which stores an array of form IDs. To manage input for this attribute, I also introduced the MultiSelectComponent which, inside of the Inspector is able to provide a searchable multi-select input. To gather the options for that input, I replaced withSelect for the useSelect hook, and select post with the type `give_forms`.

_A handful of visual design issues are still present in the MultiSelectControl this PR introduces, but until we have high fidelity designs I have opted not to address them as part of this PR. These design issues felt like a potential rabbit hole, and should be dealt with in a separate issue._

## Affects
This PR affects the Milestone block registration, as well as the Milestone Inspector rendering logic.

## Visuals
![ConnectForms](https://user-images.githubusercontent.com/5186078/92024901-a4025800-ed13-11ea-91fe-e1646f4d883e.gif)

## Pre-review Checklist

## Testing Instructions
1. Create a new post or page
2. Add the Milestone block
3. Add or remove Forms in the block settings
4. Save and refresh the page
5. Check that Forms setting persisted correctly
